submodules/misc/misc.js

```diff
@@ -1619,6 +1801,75 @@ define(function(require) {
 					callback && callback(data.data);
 				}
 			});
+		},
+
+		miscSchemasList: function(callback) {
+			var self = this;
+
+			self.callApi({
+				resource: 'schemas.list',
+				success: function(data, status) {
+					callback && callback(data.data);
+				}
+			});
+		},
+
+		miscGetSchema: function(args) {
+			var self = this,
+				data = args.data;
+
+			self.callApi({
+				resource: 'schemas.get',
+				data: {
+					schemaId: data.schemaId
+				},
+				success: function(data, status) {
+					args.hasOwnProperty('success') && args.success(data.data);
+				},
+				error: function(parsedError) {
+					args.hasOwnProperty('error') && args.error(parsedError);
+				}
+			});
+		},
+
+		/*@param reflist: array of the references to other schemas
+		*@param parentSchema: schema with references to other schemas
+		*@param jsoneditor: function to set schema in jsoneditor whith its references
+		fetch all subschemas of partenSchema so they can be set on the jsoneditor*/
+		miscGetSubSchema: function(refList, parentSchema, jsoneditor) {
+			var self = this;
+
+			monster.series(refList.map(ref =>
+				function(callback) {
+					self.miscGetSchema({
+						data: {
+							schemaId: ref
+						},
+						success: function(data) {
+						/*passing array with id,data pair to return on result callback*/
+							callback(null, [data.id, data]);
+						}
+					});
+				},
+			),	function(err, results) {
+				/*convert results array of arrays to object
+					and setting it on the jsoneditor alongside its
+					parent schema*/
+				jsoneditor.setSchema(parentSchema, Object.fromEntries(results));
```

Comment:
Since you are switching to using `monster#parallel`, you can reformat its output to be an object of functions, as the resulting object will follow the same structure and each method is replaced by the API response.

```js
monster.parallel(_
    .chain(refList)
    .keyBy()
    .mapValues(function(ref) {
        // api call
    })
    .value()
, function(err, results) {
    jsoneditor.setSchem(parentSchema, results);
})
```
^^ hopefully that clarifies things, definitely let me know if it doesn't!